Connected System Template (CST) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/helloworld/templates/DiagnosticsFactory.java

```diff
@@ -0,0 +1,59 @@
+package com.mycorp.helloworld.templates;
+
+import static com.mycorp.helloworld.templates.GoogleDriveSampleConnectedSystemTemplate.CLIENT_ID_KEY;
+import static com.mycorp.helloworld.templates.GoogleDriveSampleConnectedSystemTemplate.CLIENT_SECRET_KEY;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.appiancorp.connectedsystems.simplified.sdk.configuration.SimpleConfiguration;
+import com.google.api.services.drive.model.File;
+
+public class DiagnosticsFactory {
+
+  protected SimpleConfiguration connectedSystemConfiguration;
+  protected SimpleConfiguration integrationConfiguration;
+
+  protected Map<String,Object> requestDiagnostic = new HashMap<>();
+  protected Map<String,String> responseDiagnostic = new HashMap<>();
+
+  Map<String,Object> result = new HashMap<>();
+
+  public DiagnosticsFactory(SimpleConfiguration connectedSystemConfiguration, SimpleConfiguration integrationConfiguration) {
+    this.connectedSystemConfiguration = connectedSystemConfiguration;
+    this.integrationConfiguration = integrationConfiguration;
+  }
+
+  public Map<String,Object> getRequestDiagnostic() {
+    return requestDiagnostic;
+  }
+
+  public Map<String,String> getResponseDiagnostic() {
+    return responseDiagnostic;
+  }
+
+  public Map<String,Object> getResult() {
+    return result;
+  }
+
+  protected void addRequestDiagnostics() {
```

Comment:
This should return a map. This is a stateful object but it does not need to be stateful. You are forced to do `addRequestDiagnostic`, then `getRequestDiagnostic`.